### PR TITLE
feat: add retention policies for topic level

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

### CODE REVIEW

File: pulsar/resource_pulsar_topic.go [hunk 1]
```diff
@@ -165,6 +195,23 @@ func resourcePulsarTopicRead(d *schema.ResourceData, meta interface{}) error {
...
+
+			ret, err := client.GetRetention(*topicName, true)
+			if err != nil {
+				return fmt.Errorf("ERROR_READ_TOPIC: GetRetention: %w", err)
+			}
+
+			_ = d.Set("retention_policies", schema.NewSet(retentionPoliciesToHash, []interface{}{
+				map[string]interface{}{
+					"retention_minutes":    fmt.Sprint(ret.RetentionTimeInMinutes),
+					"retention_size_in_mb": fmt.Sprint(ret.RetentionSizeInMB),
```
Comment (codelipenghui) [inline pulsar/resource_pulsar_topic.go:209]: Looks `retention_size` and `retention_time` is more reasonable here, looks like 

```
retention_size = "1G"
retention_time = "7d"
```

Comment (nodece) [reply pulsar/resource_pulsar_topic.go:209]: It same with `pulsar_namespace`, should we update the `pulsar_namespace` config?

Comment (nodece) [reply pulsar/resource_pulsar_topic.go:209]: I tried this way, this data conversion is difficult. We can set `1G` in Terraform configuration file and parse this value into Pulsar, when we get this policy from the pulsar, `1024` is returned, which is used to apply to Terraform configuration file, but we cannot determine the suffix, so it is difficult.

File: pulsar/resource_pulsar_topic.go [hunk 2]
```diff
@@ -86,6 +86,23 @@ func resourcePulsarTopic() *schema.Resource {
 					},
 				},
 			},
+			"retention_policies": {
```
Comment (RobertIndie) [inline pulsar/resource_pulsar_topic.go:89]: Do we need to set the `Set` function for the retention_policies set here? like `Set: retentionPoliciesToHash`

File: pulsar/resource_pulsar_topic.go [hunk 3]
```diff
@@ -86,6 +86,23 @@ func resourcePulsarTopic() *schema.Resource {
...
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"retention_minutes": {
```
Comment (RobertIndie) [inline pulsar/resource_pulsar_topic.go:95]:
```suggestion
						"retention_time_minutes": {
```
Here seems to be a small name problem in both topic and namespace.

**Review by RobertIndie — CHANGES_REQUESTED** [review]

File: .github/workflows/acctest.yaml [hunk 4]
```diff
@@ -16,18 +16,14 @@ jobs:
       - name: Check out code into the Go module directory
         uses: actions/checkout@v2
 
-      - name: Get dependencies
+      - name: Install terraform@0.12.17
         run: |
           wget https://releases.hashicorp.com/terraform/0.12.17/terraform_0.12.17_linux_amd64.zip
```
[the remark this answers is not in the document]
Comment (nodece) [reply .github/workflows/acctest.yaml:21]: We should bump `terraform` to latest version, the `terraform-plugin-sdk` also too. We can do it in the next PR, #30 do the same things.

File: pulsar/resource_pulsar_topic.go (hunk 2 above)
Comment (nodece) [reply pulsar/resource_pulsar_topic.go:89]: It is unnecessary, retention_policies only has an item.

File: pulsar/resource_pulsar_topic.go (hunk 3 above)
Comment (nodece) [reply pulsar/resource_pulsar_topic.go:95]: It is same with `namespace`.

Comment (RobertIndie) [reply pulsar/resource_pulsar_topic.go:95]: Yes, the ones in `namespace` should be wrong too.

Comment (nodece) [reply pulsar/resource_pulsar_topic.go:95]: I knew how to fix this, thanks.

File: pulsar/resource_pulsar_topic.go (hunk 1 above)
Comment (codelipenghui) [reply pulsar/resource_pulsar_topic.go:209]: @nodece I see. It's not a big problem.

**Review by RobertIndie — APPROVED** [review]
LGTM

**Pull request merged**